crates/rolldown/src/module_finalizers/scope_hoisting/mod.rs

```diff
@@ -475,6 +475,8 @@ impl<'me, 'ast> ScopeHoistingFinalizer<'me, 'ast> {
     &self,
     expr: &mut ast::NewExpression<'ast>,
   ) -> Option<()> {
+    let &rec_idx = self.ctx.module.new_url_references.get(&expr.span())?;
+    let rec = &self.ctx.module.import_records[rec_idx];
```

Comment:
If we cannot obtain `rec_idx` from the map, then there is no need to perform any further pattern matching.